jubatus/core/nearest_neighbor/lsh_benchmark.cpp

```diff
@@ -0,0 +1,196 @@
+#include <vector>
+#include <string>
+#include <iostream>
+#include "lsh_function.hpp"
+#include "bit_vector_ranking.hpp"
+#include "nearest_neighbor_factory.hpp"
+#include "nearest_neighbor_base.hpp"
+#include "../common/type.hpp"
+#include "../common/jsonconfig.hpp"
+#include "../fv_converter/datum.hpp"
+#include "../storage/bit_vector.hpp"
+#include "../storage/column_type.hpp"
+#include "../storage/column_table.hpp"
+#include "../storage/abstract_column.hpp"
+
+#include "jubatus/util/lang/cast.h"
+#include "jubatus/util/lang/shared_ptr.h"
+#include "jubatus/util/math/random.h"
+#include "jubatus/util/system/time_util.h"
+
+using jubatus::util::system::time::clock_time;
+using jubatus::util::system::time::get_clock_time;
+using jubatus::util::lang::lexical_cast;
+using jubatus::util::lang::shared_ptr;
+using jubatus::core::nearest_neighbor::cosine_lsh;
+using jubatus::core::nearest_neighbor::create_nearest_neighbor;
+using jubatus::core::nearest_neighbor::nearest_neighbor_base;
+using jubatus::core::common::sfv_t;
+using jubatus::core::common::jsonconfig::config;
+using jubatus::core::storage::column_table;
+using jubatus::core::storage::column_type;
+using jubatus::core::storage::bit_vector;
+using jubatus::core::storage::bit_vector_column;
+using jubatus::core::nearest_neighbor::bit_vector_ranker;
+using jubatus::util::text::json::json;
+using jubatus::util::text::json::json_null;
+using jubatus::util::text::json::json_integer;
+using jubatus::util::text::json::json_object;
+using jubatus::util::text::json::to_json;
+
+namespace {
+sfv_t generate_sfv(size_t dencity, jubatus::util::math::random::mtrand& rand) {
+  sfv_t d;
+  d.reserve(dencity);
+  for (size_t i = 0; i < dencity; ++i) {
+    d.push_back(
+      std::make_pair(lexical_cast<std::string>(i),
+                     rand.next_double()));
+  }
+  return d;
+}
+
+std::vector<sfv_t> generate_dataset(size_t num, size_t dencity) {
+  // prepare dataset
+  std::vector<sfv_t> data;
+  data.reserve(num);
+  jubatus::util::math::random::mtrand rand;
+  for (size_t i = 0; i < num; ++i) {
+    data.push_back(generate_sfv(dencity, rand));
+  }
+  return data;
+}
+
+bit_vector random_bit_vector(size_t bit_num) {
```

Comment:
gcc warning: ‘jubatus::core::storage::bit_vector {anonymous}::random_bit_vector(size_t)’ defined but not used [-Wunused-function]